File: fab-core/src/main/java/eu/stratosphere/fab/core/beans/StringToLifespan.java
package eu.stratosphere.fab.core.beans;

import eu.stratosphere.fab.core.beans.system.Lifespan;
import org.springframework.core.convert.converter.Converter;
import scala.Enumeration.Value;

public class StringToLifespan implements Converter<String, Value> {

    @Override
    public Value convert(String s) {
        switch (s) {
            case "SUITE":
                return Lifespan.SUITE();
            case "EXPERIMENT":
                return Lifespan.EXPERIMENT();
            default:
                throw new IllegalArgumentException(s + " can not be converted to Scala Lifecycle Value!");
        }
    }

}

File: peel-core/src/main/scala/eu/stratosphere/peel/core/beans/StringToLifespan.scala
package eu.stratosphere.peel.core.beans

import eu.stratosphere.peel.core.beans.system.Lifespan
import org.springframework.core.convert.converter.Converter

class StringToLifespan extends Converter[String, Lifespan.Value]{

  def convert(s: String): Lifespan.Value = {
    if (s == "PROVIDED") {
      Lifespan.PROVIDED
    }
    else if (s == "SUITE") {
      Lifespan.SUITE
    }
    else if (s == "EXPERIMENT") {
      Lifespan.EXPERIMENT
    }
    else {
      throw new IllegalArgumentException(s + " can not be converted to Scala Lifecycle Value!")
    }
  }
}
